server/world/world.go

```diff
@@ -801,6 +802,30 @@ func (w *World) SetSpawn(pos cube.Pos) {
 	}
 }
 
+// PlayerSpawn returns the location of the player spawn in the world.
+func (w *World) PlayerSpawn(uuid uuid.UUID) cube.Pos {
+	if w == nil {
+		return cube.Pos{}
+	}
+	pos, exist, err := w.conf.Provider.LoadPlayerSpawnPosition(uuid)
+	if !exist {
+		pos = w.Spawn()
+	}
+	if err != nil {
+		w.conf.Log.Debugf("failed to get player spawn: %v", err)
+	}
+	return pos
+}
+
+// SetPlayerSpawn sets the spawn of the player. If the player has a player spawn in the world, the player will
+// be teleported to this location on respawn.
+func (w *World) SetPlayerSpawn(uuid uuid.UUID, pos cube.Pos) {
+	if w == nil {
+		return
+	}
+	w.conf.Log.Debugf("failed to set player spawn: %v", w.conf.Provider.SavePlayerSpawnPosition(uuid, pos))
```

Comment:
This will always debug this message. I would first call SavePlayerSpawnPosition and only log the error if the error is non-nil. Also, I'd recommend using `Log.Errorf` for this!